PLUGINS.md

```diff
@@ -1,54 +1,153 @@
+
+Currently, you cannot add any platform-specific resources other than `AndroidManifest.xml` and `Info.plist` in the `platforms` directory of the plugin.
+
+### Package.json Specification
+
+Every NativeScript plugin should contain a valid `package.json` file in its root. This `package.json` file must meet the following requirements.
+
+* It must comply with the [npm specification](https://docs.npmjs.com/files/package.json).<br/>The `package.json` must contain at least `name` and `version` pairs. You will later use the plugin in your code by requiring it by its `name`.
+* It must contain a `nativescript` section which describes the supported NativeScript runtimes and their versions. This section can be empty. If you want to define supported platforms and runtimes, you can nest a `platforms` section. In this `platforms` section, you can nest `ios` and `android` key-value pairs. The values in these pairs must be valid runtime versions.
+* If the plugin depends on other npm modules, it must contain a `dependencies` section as described [here](https://docs.npmjs.com/files/package.json#dependencies).<br/>The NativeScript CLI will resolve the dependencies during the plugin installation. 
 
-Here is an example. 
+#### Package.json Example
 
-Let you say that you have the following `Android.manifest` file in `platforms\android\` folder in your plugin:
+The following is an example of a `package.json` file for a NativeScript plugin which supports the 1.0.0 version of the iOS runtime and the 1.1.0 version of the Android runtime.
+
+```
+{
+  "name": "myplugin",
+  "version": "0.0.1",
+  "nativescript": {
+    "platforms": {
+      "ios": "1.0.0",
+      "android": "1.1.0"
+    }
+  }
+}
+```
+
+## Install a Plugin
+
+To install a plugin for your project, inside your project, run the following command.
+
+```Shell
+tns plugin add <Plugin>
 ```
+
+### Valid Plugin Sources
+
+You can specify a plugin by name in the npm registry, local path or URL. The following are valid values for the `<Plugin>` attribute.
+
+* A `<Name>` or `<Name>@<Version>` for plugins published in the npm registry.
+* A `<Local Path>` to the directory which contains the plugin files and its `package.json` file.
+* A `<Local Path>` to a `.tar.gz` archive containing a directory with the plugin and its `package.json` file.
+* A `<URL>` which resolves to a `.tar.gz` archive containing a directory with the plugin and its `package.json` file.
+* A `<git Remote URL>` which resolves to a `.tar.gz` archive containing a directory with the plugin and its `package.json` file.
+
+### Installation Specifics
+
+The installation of a NativeScript plugin mimics the installation of an npm module.
```

Comment:
I beg to differ here. Articles are based on how the abbreviation is read. In this case, it's en-pi-em which calls for an "an" in front.